upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -1321,7 +1321,7 @@ func setupDNSTopology(opt *NewClusterOptions, cluster *api.Cluster) error {
 	switch strings.ToLower(opt.DNSType) {
 	case "":
 		switch cluster.Spec.GetCloudProvider() {
-		case api.CloudProviderHetzner, api.CloudProviderDO:
+		case api.CloudProviderHetzner, api.CloudProviderDO, api.CloudProviderAzure:
```

Comment:
We should move most, if not all cloud providers to `dns=none` soon.
Gossip is a much worse choice and Public DNS is only available on a few cloud providers.